discussion_grading/discussion_grading.py

```diff
@@ -44,35 +153,223 @@ def resource_string(self, path: str) -> str:
 
         # Add i18n js
-        statici18n_js_url = self._get_statici18n_js_url()
-        if statici18n_js_url:
-            frag.add_javascript_url(
-                self.runtime.local_resource_url(self, statici18n_js_url)
-            )
+        if statici18n_js_url := self._get_statici18n_js_url():
+            frag.add_javascript_url(self.runtime.local_resource_url(self, statici18n_js_url))
 
+        context = {
+            "block": self,
+            "weighted_score": self.get_weighted_score(),
+        }
+
+        frag.add_content(self.render_template("static/html/discussion_grading.html", context))
+        frag.add_css(self.resource_string("static/css/discussion_grading.css"))
         frag.add_javascript(self.resource_string("static/js/src/discussion_grading.js"))
         frag.initialize_js("XBlockDiscussionGrading")
         return frag
 
+    def get_weighted_score(self) -> int:
+        """
+        Return weighted score for the current user.
+
+        Returns:
+            int | None: The weighted score.
+        """
+        score = get_score(self.get_student_item_dict())
+        return score.get("points_earned") if score else 0
+
+    def get_student_item_dict(self) -> dict:
+        """
+        Return dict required by the submissions app.
+
+        This allows creating and retrieving submissions for a particular student.
+
+        Returns:
+            dict: The student item dict.
+        """
+        return {
+            "student_id": get_anonymous_user_id(self.current_user),
+            "course_id": self.block_course_id,
+            "item_id": self.block_id,
+            "item_type": ITEM_TYPE,
+        }
+
+    def set_score(self) -> None:
+        """
+        Set the score for the current user.
+        """
+        set_score(self.submission_uuid, round(self.raw_score * self.weight), self.weight)
+
+    def create_submission(self, user_stats: dict) -> None:
+        """
+        Get the submission for the current user.
+        """
+        submission_data = create_submission(self.get_student_item_dict(), user_stats)
+        self.submission_uuid = submission_data.get("uuid")
+
+    def get_score(self, user_stats: dict) -> int:
+        """
+        Get the grade for the current user based on the grading method and number of participations.
+
+        Args:
+            user_stats (dict): The number of participations for the current user.
+
+        Returns:
+            int: The grade for the current user.
+        """
+        number_of_participations = sum(user_stats.values())
+
+        if number_of_participations >= self.number_of_participations:
+            return 1
+
+        if self.grading_method == DiscussionGradingMethod.MINIMUM_PARTICIPATIONS.name:
+            return int(number_of_participations >= self.number_of_participations)
+        elif self.grading_method == DiscussionGradingMethod.AVERAGE_PARTICIPATIONS.name:
+            return number_of_participations / self.number_of_participations
```

Comment:
I thought the submissions API didn't accept non-integer values. I don't quite remember, though.

Reply:
Should the responsibility of returning the score already rounded lay on this method?

Reply:
This must be done in the `set_score`, since it is multiplied by the weight.